pkg/agent/endpoints/sdsv3/handler_test.go

```diff
@@ -57,7 +58,7 @@ var (
 		},
 	}
 
-	fedBundle = bundleutil.BundleFromRootCA("spiffe://otherdomain.test", &x509.Certificate{
+	fedBundle = bundleutil.BundleFromRootCA(spiffeid.RequireTrustDomainFromString("spiffe://otherdomain.test"), &x509.Certificate{
```

Comment:
nit: even though RequireTrustDomainFromString can parse the trust domain out of the ID, i think it is clearer in intent to just pass the trust domain:
```suggestion
	fedBundle = bundleutil.BundleFromRootCA(spiffeid.RequireTrustDomainFromString("otherdomain.test"), &x509.Certificate{
```